datafusion/src/logical_plan/builder.rs

```diff
@@ -1162,6 +1164,117 @@ pub(crate) fn expand_qualified_wildcard(
     expand_wildcard(&qualifier_schema, plan)
 }
 
+/// check whether the logical plan we are building is valid
```

Comment:
As a follow on PR, it would be awesome to encapsulate this `LogicalPlan` tree walking code into some sort of visitor.